draft-yasskin-webpackage-use-cases.md

```diff
@@ -372,6 +372,45 @@ Associated requirements:
   origin.
 * {{transfer-compression}}{:format="title"}
 
+### Least Authority CDN Serving {#least-authority-cdn-serving}
+
+Currently, a CDN trusted via TLS to serve content for an FQDN may also serve
+arbitrary content for that FQDN. Given the distributed deployment of points of
+presence (POPs) -- often spanning many datacenters and jurisdictions -- this
+creates a security risk for the manipulation of content delivered from those
+POPs. Yet, in many CDN deployments, POPs merely distribute content generated by
+the origin without alteration. For these deployments, it would be useful to
+remove the authority of the CDN to serve arbitrary content.
+
+SXG offers a means to reducing CDN authority via HTTP clients configured to
+expect SXG for all responses, regardless of other trust. Applicable clients
+might include package managers, native mobile applications, desktop
+applications, and server applications.
+
+SXG would also offer a foundation for dynamically imposing such client
+behavior using a mechanism similar to HSTS. This would allow supporting clients
```

Comment:
I wonder if Must-Staple is a closer analogy to what is required. An origin could not enforce that their CDN includes a certain HTTP header on all responses, but they could enforce that the only certificate they bestow upon the CDN has a given extension. (As to whether that extension could or should be marked critical, I'm not sure.) Supporting clients would, when confronted with a TLS connection backed by such a cert, only honor responses that are signed with an SXG cert.

Reply:
> I wonder if Must-Staple is a closer analogy to what is required.

Perhaps I should rephrase the text, as the forward-looking idea is less about the mechanism (HSTS-style, Must-Staple, or otherwise) and more about allowing a site to inform clients to expect SXG responses. That expectation should be sticky or, at least, resistant to downgrade attacks.

I agree that an extension applied to the TLS certificate would be a technologically elegant way to solve the issue. I'd argue that such an extension should be non-critical, as that would allow existing clients to interact as before, while clients aware of the extension could enforce SXG on responses. SXG enforcement is a defense-in-depth against POP/proxy compromise, which I think is another argument for a non-critical designation.

I try to push back against new certificate extensions for practical reasons, though. They create a barrier to entry for people who implement or deploy the design. That raises some reasons I looked first to HSTS: it's broadly used, doesn't require external facilitation (e.g. by a CA), and has no up-front or ongoing costs to web site administrators.

> An origin could not enforce that their CDN includes a certain HTTP header on all responses, but they could enforce that the only certificate they bestow upon the CDN has a given extension.

This is true, but what I had in mind was the broader world of HSTS rather than just the header. In practice, qualifying HSTS headers get cached in browsers and bundled into the HSTS Preload List, both of which continue enforcement even if the header goes missing. (So, while a compromised POP could stop sending an SXG enforcement header, clients would still expect SXG.) I'm unsure whether this is the most desirable approach to SXG enforcement, but it's one established pattern.